server/utils/crmDisplayHelper.ts

```diff
@@ -1,12 +1,75 @@
 import { CrmResponse } from '@eqApi'
-import { Section, ShowOrHideWhen } from '@crmDisplay'
+import { Section, ShowOrHideWhen, Subsection } from '@crmDisplay'
 import _ from 'lodash'
 
 export const includeSection = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
   if (section.hideWhen && conditionsMet(section.hideWhen, crmResponse)) {
     return false
   }
-  return !section.showWhen || conditionsMet(section.showWhen, crmResponse)
+
+  const include = !section.showWhen || conditionsMet(section.showWhen, crmResponse)
+
+  if (include && isSectionEmpty(section, crmResponse)) {
+    return false
+  }
+
+  return include
+}
+
+// Utility to check if a section is empty
+const isSectionEmpty = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
+  if (!section.subsections || section.subsections.length === 0) {
+    return false
+  }
+
+  return section.subsections.every(subsection => isSubsectionEmpty(subsection, crmResponse))
+}
+
+export const isSubsectionEmpty = <T extends CrmResponse>(subsection: Subsection, crmResponse: T): boolean => {
```

Comment:
Only used in this file, so export can be removed
```suggestion
const isSubsectionEmpty = <T extends CrmResponse>(subsection: Subsection, crmResponse: T): boolean => {
```